code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm

```diff
@@ -0,0 +1,367 @@
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/proc/check_enraged()
+	if(enraged)
+		return
+	if(health > maxHealth*0.25)
+		return
+	SetRecoveryTime(80, 80)
+	adjustHealth(-maxHealth)
+	enraged = TRUE
+	enraging = TRUE
+	animate(src, pixel_y = pixel_y + 96, time = 100, easing = ELASTIC_EASING)
+	spin(100, 10)
+	SLEEP_CHECK_DEATH(60)
+	playsound(src, 'sound/effects/explosion3.ogg', 100, TRUE)
+	icon_state = "demonic_miner_phase2"
+	animate(src, pixel_y = pixel_y - 96, time = 8, flags = ANIMATION_END_NOW)
+	spin(8, 2)
+	SLEEP_CHECK_DEATH(8)
+	for(var/mob/living/L in viewers(src))
+		shake_camera(L, 3, 2)
+	playsound(src, 'sound/effects/meteorimpact.ogg', 100, TRUE)
+	setMovetype(movement_type | FLYING)
+	enraging = FALSE
+	adjustHealth(-maxHealth)
+
+/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/death(gibbed, list/force_grant)
+	if(health > 0)
+		return
+	var/turf/T = get_turf(src)
+	var/loot = rand(1, 3)
+	switch(loot)
+		if(1)
+			new /obj/item/resurrection_crystal(T)
+		if(2)
+			new /obj/item/clothing/shoes/winterboots/ice_boots/speedy(T)
+		if(3)
+			new /obj/item/pickaxe/drill/jackhammer/demonic(T)
+	return ..()
+
+/obj/item/resurrection_crystal
+	name = "resurrection crystal"
+	desc = "When used by anything holding it, this crystal gives them a second chance at life if they die."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "demonic_crystal"
+
+/obj/item/resurrection_crystal/attack_self(mob/living/user)
+	if(!iscarbon(user))
+		to_chat(user, "<span class='notice'>A dark presence stops you from absorbing the crystal.</span>")
+		return
+	forceMove(user)
+	to_chat(user, "<span class='notice'>You feel a bit safer... but a demonic presence lurks in the back of your head...</span>")
+	RegisterSignal(user, COMSIG_MOB_DEATH, .proc/resurrect)
+
+/// Resurrects the target when they die by cloning them into a new duplicate body and transferring their mind to the clone on a safe station turf
+/obj/item/resurrection_crystal/proc/resurrect(mob/living/carbon/user, gibbed)
+	user.visible_message("<span class='notice'>You see [user]'s soul dragged out of their body!</span>", "<span class='notice'>You feel your soul dragged away to a fresh body!</span>")
+	var/typepath = user.type
+	var/turf/T = find_safe_turf()
+	var/mob/living/carbon/clone = new typepath(T)
+	clone.real_name = user.real_name
+	user.dna.transfer_identity(clone)
+	clone.updateappearance(mutcolor_update=1)
+	user.mind.transfer_to(clone) // second life
+	to_chat(clone, "<span class='notice'>You blink and find yourself in [get_area_name(T)].</span>")
+	user.gib()
+	qdel(src)
+
+/obj/item/clothing/shoes/winterboots/ice_boots/speedy
+	name = "cursed ice hiking boots"
+	desc = "A pair of winter boots contractually made by a devil, they cannot be taken off once put on."
+	slowdown = SHOES_SPEED_SLIGHT
+
+/obj/item/clothing/shoes/winterboots/ice_boots/speedy/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT)
```

Comment:
er, can you put these on if you pick them up from the ground, or will they be stuck to your hand?

Reply:
you can put them on but they will be stuck to your hand same as eye of god